include/lsst/meas/extensions/shapeHSM/HsmShapeControl.h

```diff
@@ -11,62 +38,147 @@ enum MeasType {
     SHEAR,                              // We will measure g1,g2
 };
 
-class HsmShapeControl : public algorithms::AlgorithmControl {
+/**
+ *  @brief A C++ control class to handle HsmShapeAlgorithm's configuration
+ *
+ *  In C++, we define Control objects to handle configuration information.  Using the LSST_CONTROL_FIELD
+ *  macro and lsst.pex.config.wrap.makeConfigClass, we can turn these into more full-featured Config classes
+ *  in Python.  While the user will usually interact with the Config class, the plugin wrapper system will
+ *  turn Config instances into Control instances when passing them to C++.
+ *
+ *  This should logically be an inner class, but Swig doesn't know how to parse those.
+ */
+class HsmShapeControl {
 public:
-    LSST_CONTROL_FIELD(badMaskPlanes, std::vector<std::string>, "Mask planes used to reject bad pixels.");
     LSST_CONTROL_FIELD(deblendNChild, std::string, "Field name for number of deblend children");
-protected:
+    LSST_CONTROL_FIELD(badMaskPlanes, std::vector<std::string>, "Mask planes that indicate pixels that should be excluded from the fit");
 
-    HsmShapeControl(HsmShapeControl const & other) :
-        AlgorithmControl(other), badMaskPlanes(other.badMaskPlanes), deblendNChild(other.deblendNChild) {}
+    /**
+     *  @brief Default constructor
+     *
+     *  All control classes should define a default constructor that sets all fields to their default values.
+     */
+    HsmShapeControl() : _name("") {
+        badMaskPlanes.push_back("BAD");
+        badMaskPlanes.push_back("SAT");
+        badMaskPlanes.push_back("INTRP");
+    }
 
-    HsmShapeControl(std::string const & name) : AlgorithmControl(name, 1.0), deblendNChild("") {
+    HsmShapeControl(std::string const & name) : _name(name) {
         badMaskPlanes.push_back("BAD");
         badMaskPlanes.push_back("SAT");
         badMaskPlanes.push_back("INTRP");
     }
+
+private:
+    std::string _name;
 };
 
 class HsmShapeBjControl : public HsmShapeControl {
 public:
-    HsmShapeBjControl() : HsmShapeControl("shape.hsm.bj") {}
-private:
-    virtual PTR(algorithms::AlgorithmControl) _clone() const;
-    virtual PTR(algorithms::Algorithm) _makeAlgorithm(
-        afw::table::Schema & schema, PTR(daf::base::PropertyList) const & metadata
-    ) const;
+    HsmShapeBjControl() : HsmShapeControl("extensions_shapeHSM_HsmShapeBj") {}
```

Comment:
As I recommended with Kron, let's shorted "extensions" to just "ext" in these names.